mobile_config_cli/src/client.rs

```diff
@@ -142,6 +152,36 @@ impl AuthClient {
     }
 }
 
+impl CarrierClient {
+    pub async fn new(host: &str, server_pubkey: &str) -> Result<Self> {
+        Ok(Self {
+            client: carrier_service_client::CarrierServiceClient::connect(host.to_owned()).await?,
+            server_pubkey: PublicKey::from_str(server_pubkey)?,
+        })
+    }
+
+    pub async fn list_incentive_promotions(
+        &mut self,
+        keypair: &Keypair,
+    ) -> Result<CarrierIncentivePromotionListResV1> {
+        let timestamp = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs();
+        let mut request = CarrierIncentivePromotionListReqV1 {
+            timestamp,
+            signer: keypair.public_key().into(),
+            signature: vec![],
+        };
```

Comment:
is this list proprietary such that we need to be signature-gating a request?

Reply:
yes, only the mobile-verifier should be retrieving it.

Reply:
If that’s the case why did we add it to the cli and not just the client?